Search stories

maddiecomtois opened the pull request:
- Search stories by their title or their content text
- Code same as that used in the admin component ```find-user``` and the pipe ```filter.pipe.ts```
- Need to add translation for "Search story title or content"
- Should we make the overall page bigger/get rid of book image? 
- Should we add autofocus to the search bar?  I found it a bit distracting

<img width="802" alt="Captura de Pantalla 2021-08-16 a la(s) 13 21 08" src="https://user-images.githubusercontent.com/13711456/129611163-31cfaaf4-d4d6-480a-a7b1-5c2e484ef7b0.png">
<img width="708" alt="Captura de Pantalla 2021-08-16 a la(s) 13 22 09" src="https://user-images.githubusercontent.com/13711456/129611285-93898dcf-5648-433f-b564-eb8d681f54ff.png">
<img width="711" alt="Captura de Pantalla 2021-08-16 a la(s) 13 21 52" src="https://user-images.githubusercontent.com/13711456/129611244-8797cdfa-5380-47d1-ae8e-b6f92a7a9195.png">

NeasaNi commented:
Translation for "Search story title or content":  "Cuardaigh teideal nó ábhar scéil"

I'd be in favour of making the overall page bigger and if that means dropping the book image we can live with that...

Agree with autofocus sometimes being distracting! Images above look great.

maddiecomtois commented:
Changed some css to make the contents bigger.  Do we want it bigger? (However I'm having issues with making the height bigger.  Maybe someone who is good with css would be able to make better adjustments.). Also added the translation.

<img width="1780" alt="Captura de Pantalla 2021-08-17 a la(s) 13 35 31" src="https://user-images.githubusercontent.com/13711456/129781820-01a3d08f-ddd5-4921-b078-2238e959d947.png">

OisinNolan reviewed (approved):
Awesome feature!! Happy to merge as-is.

Left one suggestion on the filter pipe.

maddiecomtois commented on ngapp/src/app/pipes/filter.pipe.ts:
```diff
@@ -21,8 +21,12 @@ export class FilterPipe implements PipeTransform {
+    // it.username for admin, it.title for student
     return items.filter(it => {
-      return it.username.toLocaleLowerCase().includes(searchText);
+      if(it.username)
+        return it.username.toLocaleLowerCase().includes(searchText);
+      return it.title.toLocaleLowerCase().includes(searchText) || it.text.toLocaleLowerCase().includes(searchText);
```
I like this approach, it's much cleaner and easier to understand.  But I can't get it to work, seems like the object type is not passed into the filter? It doesn't recognize any of the objects being an instance of Story or User.

OisinNolan replied:
It seems like the objects we were getting from the backend (stories, users) were just being stored as objects on the frontend. I added a `fromJSON()` function to `Story` and `User` so that we can ensure that they get the correct type on the frontend - this allows us to check types in the filter function. I wanted to commit it to this branch but for some reason it merged the PR when I pushed, sorry about that! The code I added is here anyway: https://github.com/OisinNolan/An-Scealai/commit/cdf596a569288919dfcee86016ac8b4a63f46f92

I totally agree on making the window bigger and getting rid of some of the paper layers. We could have a call to discuss updating the design maybe?

maddiecomtois replied:
Looks good, probably a good thing to store object types on the front end.  I'm free Friday/next week if you want to have a call about the design

OisinNolan replied:
Awesome, sounds good. Friday afternoon suits me for a call - we can discuss in main call if there is one on Friday.

maddiecomtois commented:
What do we think about the bigger window?  Is there too much paper image everywhere? We could try to make the background white for the story titles?  Also can't seem to make the window any taller, it's kind of a small wide rectangle at the moment
